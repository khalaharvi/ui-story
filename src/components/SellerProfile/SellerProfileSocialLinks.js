import React from 'react';
import PropTypes from 'prop-types';

import { LinkWithIcon } from 'components/Link';
import {
  Facebook as FacebookIcon,
  Twitter as TwitterIcon,
  Instagram as InstagramIcon,
  Pinterest as PinterestIcon,
  NewWindow as NewWindowIcon
} from 'icons';
import { capitalizeFirstLetter } from 'utils';
import { cx } from './SellerProfile';

const icons = {
  facebook: <FacebookIcon />,
  twitter: <TwitterIcon />,
  instagram: <InstagramIcon />,
  pinterest: <PinterestIcon />,
  website: <NewWindowIcon />
};

const SellerProfileSocialLinks = ({ links }) => (
  <ul className={cx('info-social-links')}>
    {
      links.map(link => (
        <li key={link.type}>
          <LinkWithIcon
            to={link.url}
            icon={icons[link.type]}
            label={link.label || capitalizeFirstLetter(link.type)}
            external
          />
        </li>
      ))
    }
  </ul>
);

SellerProfileSocialLinks.propTypes = {
  links: PropTypes.arrayOf(PropTypes.shape({
    type: PropTypes.oneOf(['website', 'facebook', 'instagram', 'pinterest', 'twitter']).isRequired,
    url: PropTypes.string.isRequired,
    label: PropTypes.string
  })).isRequired
};

export default SellerProfileSocialLinks;
