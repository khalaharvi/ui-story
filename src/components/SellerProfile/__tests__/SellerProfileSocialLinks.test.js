import React from 'react';
import { mount, shallow } from 'enzyme';

import SellerProfileSocialLinks from '../SellerProfileSocialLinks';

const links = [
  {
    type: 'website',
    url: 'https://deringhall.com',
    label: 'Official Website'
  },
  {
    type: 'facebook',
    url: 'https://facebook.com'
  },
  {
    type: 'twitter',
    url: 'https://twitter.com'
  },
  {
    type: 'pinterest',
    url: 'https://pinterest.com'
  },
  {
    type: 'instagram',
    url: 'https://instagram.com'
  }
];

describe('<SellerProfileSocialLinks />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <SellerProfileSocialLinks links={links} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders facebook icon correctly', () => {
    const wrapper = mount(
      <SellerProfileSocialLinks
        links={[{
          type: 'facebook',
          url: 'https://facebook.com'
        }]}
      />
    );

    const link = wrapper.find('a');
    expect(link).toHaveLength(1);
    expect(link.find('Facebook')).toHaveLength(1);

    wrapper.unmount();
  });

  it('renders instagram icon correctly', () => {
    const wrapper = mount(
      <SellerProfileSocialLinks
        links={[{
          type: 'instagram',
          url: 'https://instagram.com'
        }]}
      />
    );

    const link = wrapper.find('a');
    expect(link).toHaveLength(1);
    expect(link.find('Instagram')).toHaveLength(1);

    wrapper.unmount();
  });

  it('renders pinterest icon correctly', () => {
    const wrapper = mount(
      <SellerProfileSocialLinks
        links={[{
          type: 'pinterest',
          url: 'https://pinterest.com'
        }]}
      />
    );

    const link = wrapper.find('a');
    expect(link).toHaveLength(1);
    expect(link.find('Pinterest')).toHaveLength(1);

    wrapper.unmount();
  });

  it('renders twitter icon correctly', () => {
    const wrapper = mount(
      <SellerProfileSocialLinks
        links={[{
          type: 'twitter',
          url: 'https://twitter.com'
        }]}
      />
    );

    const link = wrapper.find('a');
    expect(link).toHaveLength(1);
    expect(link.find('Twitter')).toHaveLength(1);

    wrapper.unmount();
  });

  it('renders new window icon correctly', () => {
    const wrapper = mount(
      <SellerProfileSocialLinks
        links={[{
          type: 'website',
          url: 'https://deringhall.com'
        }]}
      />
    );

    const link = wrapper.find('a');
    expect(link).toHaveLength(1);
    expect(link.find('NewWindow')).toHaveLength(1);

    wrapper.unmount();
  });
});
