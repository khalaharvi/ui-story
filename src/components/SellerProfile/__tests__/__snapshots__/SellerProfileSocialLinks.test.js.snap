// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellerProfileSocialLinks /> renders correctly 1`] = `
<ul
  className="dh-seller-profile-info-social-links"
>
  <li
    key="website"
  >
    <LinkWithIcon
      external={true}
      icon={<NewWindow />}
      label="Official Website"
      to="https://deringhall.com"
    />
  </li>
  <li
    key="facebook"
  >
    <LinkWithIcon
      external={true}
      icon={<Facebook />}
      label="Facebook"
      to="https://facebook.com"
    />
  </li>
  <li
    key="twitter"
  >
    <LinkWithIcon
      external={true}
      icon={<Twitter />}
      label="Twitter"
      to="https://twitter.com"
    />
  </li>
  <li
    key="pinterest"
  >
    <LinkWithIcon
      external={true}
      icon={<Pinterest />}
      label="Pinterest"
      to="https://pinterest.com"
    />
  </li>
  <li
    key="instagram"
  >
    <LinkWithIcon
      external={true}
      icon={<Instagram />}
      label="Instagram"
      to="https://instagram.com"
    />
  </li>
</ul>
`;
